service/storage_service.cc

```diff
@@ -400,12 +400,13 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
     // Ensure we know our own actual Schema UUID in preparation for updates
     co_await db::schema_tables::recalculate_schema_version(_sys_ks, proxy, _feature_service);
 
-    assert(_group0);
-    co_await _group0->setup_group0(_sys_ks.local(), initial_contact_nodes);
-
     app_states.emplace(gms::application_state::NET_VERSION, versioned_value::network_version());
     app_states.emplace(gms::application_state::HOST_ID, versioned_value::host_id(local_host_id));
     app_states.emplace(gms::application_state::RPC_ADDRESS, versioned_value::rpcaddress(broadcast_rpc_address));
+    if (_group0->raft_gr().is_enabled()) {
+        auto addr = co_await _group0->load_or_create_my_addr();
```

Comment:
So now we basically de-coupled raft id creation from joining group0. Not sure if this is a problem or not. It is possible to add this gossiper state only after we initiated group0 join? Probably not since we need others to know our address for "join" to complete.

Reply:
no, we need to do it before so that others can reach us with raft snapshot